### DISCUSSION TITLE
its type '() => promise<element>' is not a valid jsx element type

### DISCUSSION THREAD

**Discussion by aleynaeser** [vercel/next.js#73645] [Help]
### Summary

I encountered this error after deleting the node_modules folder and reinstalling dependencies. The project has been working fine for over a year with no errors, and I haven't made significant changes to the code.

Additional Context:

    "next": "14.2.6",
    "react": "^18",
    "react-dom": "^18",
    "typescript": "^5"
    "eslint": "^8",
    "eslint-config-next": "14.2.6",
   
   
```
Its type '({ systemCompanyAddress }: ISystemCompanyInfo) => Promise<Element>' is not a valid JSX element type.
    Type '({ systemCompanyAddress }: ISystemCompanyInfo) => Promise<Element>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'Promise<Element>' is not assignable to type 'ReactNode'.
```

**Comment by icyJoseph** [comment]
This looks like the error we were getting when Server Components where first introduced. 

- https://nextjs.org/docs/app/api-reference/config/typescript#with-async-server-componens

Could you check that you have the correct TS version? Perhaps your IDE is running another TS version than the one included into your package.json?

**Reply by aleynaeser** [reply]
I have tried the suggestions provided here but couldn’t get them to work. Do you have a chance to test this on your side?

**Reply by icyJoseph** [reply]
If you have a repository, I'd check it ASAP

**Reply by icyJoseph** [reply]
Thinking out loud here, are your React and ReactDOM types floating dependencies? I wonder if by removing your node_modules/lock file whichever, you might have ended up taking a new minor/patch?

**Comment by Robert-Langbein** [comment]
received the same error after running a npm update. Any ideas how to fix? Also worked without any problems before.

**Reply by icyJoseph** [reply]
Got a repository to see?

**Comment by Robert-Langbein** [comment]
I have absolutely no idea why this happens. It works perfectly fine in a fresh Next.js project. I can implement an async server component just like this:
`
const ServerPage = () => {
  return (
    <>
      <AsyncComponent />
    </>
  );
};
`

However, in my actual project, I managed to get it working by making the primary server page itself async and awaiting the component:
`
const ServerPage = async () => {
  return (
    <>
      {await AsyncComponent()}
    </>
  );
};
`

This feels more like a workaround, as my project is still a WIP, and I don't consider it a permanent solution. Maybe someone knows more about why this happens.

**Reply by icyJoseph** [reply]
Uuh that's not even correct at all... But let's not get lost in other details.

Care to show us package.json versions of relevant packages? What Package manager are you using?

**Reply by icyJoseph** [reply]
And where do you see the error? At build time? In your IDE/text editor, perhaps both?

**Reply by Robert-Langbein** [reply]
I'm using npm and encountering the error both at build time and in my IDE (VS Code). Here’s my current package.json:

```
{
  "name": "frontend",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "rm -rf .next/cache && next dev",
    "build": "npx prettier src --check && npx eslint src && next build",
    "start": "next start",
    "lint": "next lint",
    "check": "npx prettier src --check && npx eslint src"
  },
  "dependencies": {
    "@eslint/js": "^9.3.0",
    "@hello-pangea/dnd": "^16.6.0",
    "@types/react-window": "^1.8.8",
    "@typescript-eslint/eslint-plugin": "^7.13.0",
    "axios": "^1.3.1",
    "chart.js": "^4.4.3",
    "embla-carousel-react": "^8.1.7",
    "eslint-config-prettier": "^9.1.0",
    "globals": "^15.2.0",
    "lodash": "^4.17.21",
    "next": "15.0.4",
    "next-auth": "^4.24.7",
    "react-beautiful-dnd": "^13.1.1",
    "react-chartjs-2": "^5.2.0",
    "react-dnd": "^16.0.1",
    "react-dropzone": "14.2.3",
    "react-hot-toast": "^2.4.1",
    "react-markdown": "^9.0.1",
    "react-swipeable": "^7.0.1",
    "react-window": "^1.8.10",
    "remark-gfm": "^4.0.0",
    "typescript-eslint": "^7.9.0",
    "zustand": "^4.5.5",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "@types/lodash": "^4.17.7",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-beautiful-dnd": "^13.1.8",
    "@types/react-dom": "^18",
    "daisyui": "^4.10.2",
    "eslint": "^8",
    "eslint-config-next": "15.0.4",
    "eslint-config-standard": "^17.1.0",
    "eslint-plugin-check-file": "^2.8.0",
    "eslint-plugin-import": "^2.29.1",
    "eslint-plugin-n": "^16.6.2",
    "eslint-plugin-prettier": "^5.1.3",
    "eslint-plugin-promise": "^6.1.1",
    "postcss": "^8",
    "tailwindcss": "^3.4.1",
    "typescript": "^5"
  }
}
```

**Reply by icyJoseph** [reply]
Could you please, pin the react and react-dom types to, 18.3.13 and 18.3.1, respectively? Using floating dependencies is not very good, to say the least, in production apps.

**Reply by aleynaeser** [reply]
We solved the problem with these versions
“react”: “^18.2.0",
    “react-dom”: “^18.2.0",
    “typescript”: “^4.7.4"
“@types/node”: “^18.6.3",
    “@types/react”: “^18.0.15",
    “@types/react-dom”: “^18.0.6",

**Reply by icyJoseph** [reply]
Going back to Typescript 4.x is not great either... Are you able to take the react type dependencies I posted?

**Reply by aleynaeser** [reply]
@icyJoseph  You mean with floating dependencies this, right? ^18

We fit package nextjs 14.2.6 https://github.com/vercel/next.js/blob/v14.2.6/package.json

**Reply by icyJoseph** [reply]
Nope, fix/pin it to those exact versions. Typescript you should be able to have any 5.x stable version

**Reply by aleynaeser** [reply]
> Going back to Typescript 4.x is not great either... Are you able to take the react type dependencies I posted?

Okay, tomorrow i will try it. Do u have any idea  for its reason? React 19 was released as stable. I got the error in the first build after the release.

**Reply by icyJoseph** [reply]
Ah I misread your comment...

Yeah remove the ^, to fix the type dependencies.

Well the types packages got also updated, as React 19 was released, including the React 18 types (there's a PR to revert some of that change set)

Since you did not have a fixed dependency, your app just picked up the latest types updates and broke.
